RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -61,27 +61,97 @@ public bool Generate(ProcessedArgs args)
                 throw new ArgumentNullException("args");
             }
 
-            IPackage package = this.packageHandler.FetchPackage(args.PackageId, args.PackageVersion);
-            if (package == null)
+            IPackage targetPackage = this.packageHandler.FetchPackage(args.PackageId, args.PackageVersion);
+            IEnumerable<IPackage> dependencyPackages = this.packageHandler.GetInstalledDependencies(targetPackage);
+
+            if (targetPackage == null)
             {
                 return false;
             }
 
-            string packageDir = this.packageHandler.GetLocalPackageRootDirectory(package);
-            IEnumerable<DiagnosticAnalyzer> analyzers = GetAnalyzers(packageDir, this.packageHandler.LocalCacheRoot, args.Language);
-            if (!analyzers.Any())
+            // Check that there are analyzers in the target package from which information can be extracted
+            if (!HasAnalyzers(targetPackage, args.Language))
             {
-                return false;
+                this.logger.LogWarning(UIResources.APG_NoAnalyzersFound, targetPackage.Id);
+
+                if (!args.RecurseDependencies || !dependencyPackages.Any())
+                {
+                    return false;
```

Comment:
Usability: if the main package doesn't have any analysers and the user didn't specify _/recurse_ then it would be helpful to tell them about the _/recurse_ option at this point.